### ISSUE TITLE
There was no fonts in Linux version

### ISSUE THREAD

**Issue by karianna** [AdoptOpenJDK/openjdk-support#70]
_From @JamesMung on October 23, 2018 3:54_

JDK Version ; jdk8u181-b13

When I use jasper print PDF in Windows is fine, but for Linux found the error as follow:
```java
Caused by: net.sf.jasperreports.engine.JRRuntimeException: java.io.IOException: Problem reading font data.
        at net.sf.jasperreports.engine.fonts.SimpleFontFace.<init>(SimpleFontFace.java:108)
        at net.sf.jasperreports.engine.fonts.SimpleFontFace.<init>(SimpleFontFace.java:128)
        at net.sf.jasperreports.engine.fonts.SimpleFontFace.getInstance(SimpleFontFace.java:67)
        at net.sf.jasperreports.engine.fonts.SimpleFontFamily.setNormal(SimpleFontFamily.java:99)
        at net.sf.jasperreports.engine.fonts.SimpleFontExtensionHelper.parseFontFamily(SimpleFontExtensionHelper.java:261)
        at net.sf.jasperreports.engine.fonts.SimpleFontExtensionHelper.parseFontFamilies(SimpleFontExtensionHelper.java:232)
        at net.sf.jasperreports.engine.fonts.SimpleFontExtensionHelper.loadFontFamilies(SimpleFontExtensionHelper.java:193)
        at net.sf.jasperreports.engine.fonts.SimpleFontExtensionHelper.loadFontFamilies(SimpleFontExtensionHelper.java:162)
        at net.sf.jasperreports.engine.fonts.FontExtensionsRegistry.getExtensions(FontExtensionsRegistry.java:56)
        at net.sf.jasperreports.extensions.DefaultExtensionsRegistry.getExtensions(DefaultExtensionsRegistry.java:110)
        at net.sf.jasperreports.engine.util.JRStyledTextParser.<clinit>(JRStyledTextParser.java:83)
        ... 73 more
Caused by: java.io.IOException: Problem reading font data.
        at java.awt.Font.createFont0(Font.java:1000)
        at java.awt.Font.createFont(Font.java:877)
        at net.sf.jasperreports.engine.fonts.SimpleFontFace.<init>(SimpleFontFace.java:100)
        ... 83 more
```

Can you help for solve this problem in current version ? And please fix it in future version.

_Copied from original issue: AdoptOpenJDK/openjdk8-releases#16_

**Comment by sidvar9** [comment]
We are also having the same issue and wondering where we are in providing the re-licensed Lucida fonts. Its mentioned in the AdoptOpenJDK migration guide that these will be available by Q2 2019. Thanks!

**Comment by lumpfish** [comment]
@karianna - is there a plan for resolving this?  There are a number of AdoptOpenJDK users requesting these fonts and currently the AdoptOpenJDK position is that they are to be included in the AdoptOpenJDK builds.

I had interpreted the statement "The Lucida fonts that are available in Oracle JDK 8 have a proprietary 3rd party license. AdoptOpenJDK intends to provide relicensed Lucida fonts." to mean that the AdoptOpenJDK community would be seeking a license.  Is that correct?  If so when / how would that happen?

**Comment by aahlenst** [comment]
Would including [DejaVu](https://dejavu-fonts.github.io/) suffice? Should be easier to package because it is free.

**Comment by karianna** [comment]
> @karianna - is there a plan for resolving this? There are a number of AdoptOpenJDK users requesting these fonts and currently the AdoptOpenJDK position is that they are to be included in the AdoptOpenJDK builds.
> 
> I had interpreted the statement "The Lucida fonts that are available in Oracle JDK 8 have a proprietary 3rd party license. AdoptOpenJDK intends to provide relicensed Lucida fonts." to mean that the AdoptOpenJDK community would be seeking a license. Is that correct? If so when / how would that happen?

That's a potential path forward.  I'm not sure if there's a an OSS replacement that has the same L&F

**Comment by jerboaa** [comment]
Does this also happen with 8u252 EA and `-Dsun.java2d.renderer=sun.java2d.marlin.MarlinRenderingEngine` specified?

**Comment by karianna** [comment]
The plan is to do the release, add the missing fontconfig.properties file in the nightlies and if all goes well then release with an extra .1 version

**Comment by madhu-broad** [comment]
Hi,

I had raised the below issue and see it got closed saying the duplicate of issue 76. 

We are waiting on this.. Hope, the issue gets resolve by end of june-2020.

AdoptOpenJDK/openjdk-support Getting NULL pointer exception at sun.awt.FcFontManager.getDefaultPlatformFont(FcFontManager.java:76)#80

**Comment by aahlenst** [comment]
@karianna What are your plans here? We're very close to the next quarterly patch update (14th of July).

**Comment by karianna** [comment]
> @karianna What are your plans here? We're very close to the next quarterly patch update (14th of July).

I'll make sometime next week with @gdams and @johnoliver and see if we can quickly resolve this.

**Comment by karianna** [comment]
This may be more of an installer fix - i.e. Adding the distro's fontconfig as a dependency

**Comment by karianna** [comment]
@aahlenst - Is adding fontconfig as a dependency in our installers an option here?

**Comment by aahlenst** [comment]
@karianna Sure. But if we install fontconfig, we get freetype, too. That's bundled, however, which increases the installation size and having stuff using different versions of freetype might introduce problems, too. I'd like to remove the bundled freetype (Azul does that), but that's something that isn't properly tested by us. And then we end up with https://github.com/AdoptOpenJDK/openjdk-installer/pull/130 that's lingering around for a year or so because testing is a rabbit hole in this specific case. Besides, I assume that most people here use the binary distributions.

tl:dr: I'm looking for advice or help.

**Comment by karianna** [comment]
> @karianna Sure. But if we install fontconfig, we get freetype, too. That's bundled, however, which increases the installation size and having stuff using different versions of freetype might introduce problems, too. I'd like to remove the bundled freetype (Azul does that), but that's something that isn't properly tested by us. And then we end up with [AdoptOpenJDK/openjdk-installer#130](https://github.com/AdoptOpenJDK/openjdk-installer/pull/130) that's lingering around for a year or so because testing is a rabbit hole in this specific case. Besides, I assume that most people here use the binary distributions.
> 
> tl:dr: I'm looking for advice or help.

I'm not very good on Linux packaging but can we add fontconfig and tell it _not_ to install freetype?

**Comment by aahlenst** [comment]
> can we add fontconfig and tell it not to install freetype?

No. This would break the dependency chain.

**Comment by karianna** [comment]
Ugh, well this is a challenge.   OK, so a suggested would be to unbundle FreeType without builds and instead ask the installer to add fontconfig and therefore FreeType.  But of course, this doesn't solve the issue for folks not using an installer.  So we're back at square one.

**Comment by aahlenst** [comment]
What the others are doing (on 8)...

Liberica:

```
$ jdk8u252 -name "*freetype*"
jdk8u252/jre/lib/amd64/libfreetype.so
$ find jdk8u252 -name "*font*"    
jdk8u252/jre/lib/psfont.properties.ja
jdk8u252/jre/lib/psfontj2d.properties
jdk8u252/jre/lib/amd64/libfontmanager.so
```

Zulu:

```
$ find zulu8.46.0.19-ca-jdk8.0.252-linux_x64 -name "*freetype*"
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/amd64/libfreetype.so.6
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/amd64/libfreetype.so
$ find zulu8.46.0.19-ca-jdk8.0.252-linux_x64 -name "*font*"    
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.SuSE.bfc
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.Fedora.bfc
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/psfont.properties.ja
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.SuSE.properties.src
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.Ubuntu.bfc
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.Fedora.properties.src
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.properties.src
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/psfontj2d.properties
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/amd64/libfontmanager.so
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.bfc
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/jre/lib/fontconfig.Ubuntu.properties.src
```

In comparison, AdoptOpenJDK:

```
$ find jdk8u252-b09 -name "*freetype*"
jdk8u252-b09/jre/lib/amd64/libfreetype.so.6
$ find jdk8u252-b09 -name "*font*"    
jdk8u252-b09/jre/lib/psfont.properties.ja
jdk8u252-b09/jre/lib/psfontj2d.properties
jdk8u252-b09/jre/lib/amd64/libfontmanager.so
```

Running the test program in https://github.com/AdoptOpenJDK/openjdk-support/issues/70#issuecomment-481683741, Liberica fails in the same way as AdoptOpenJDK. Azul Zulu performs slightly better, in that sense that the error message is meaningful:

```
zulu8.46.0.19-ca-jdk8.0.252-linux_x64/bin/java FontTest
Exception in thread "main" java.lang.Error: Probable fatal error:No fonts found.
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1236)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.findOtherDeferredFont(SunFontManager.java:1037)
	at sun.font.SunFontManager.findDeferredFont(SunFontManager.java:1054)
	at sun.font.SunFontManager.findFont2D(SunFontManager.java:2256)
	at sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1220)
	at sun.font.SunFontManager.initialiseDeferredFont(SunFontManager.java:1100)
	at sun.font.SunFontManager.initialiseDeferredFonts(SunFontManager.java:927)
	at sun.font.SunFontManager.loadFonts(SunFontManager.java:3302)
	at sun.awt.X11FontManager.loadFonts(X11FontManager.java:466)
	at sun.font.SunFontManager.loadFontFiles(SunFontManager.java:3429)
	at sun.font.SunFontManager.getInstalledFontFamilyNames(SunFontManager.java:3766)
	at sun.java2d.SunGraphicsEnvironment.getAvailableFontFamilyNames(SunGraphicsEnvironment.java:225)
	at sun.java2d.SunGraphicsEnvironment.getAvailableFontFamilyNames(SunGraphicsEnvironment.java:252)
	at sun.java2d.HeadlessGraphicsEnvironment.getAvailableFontFamilyNames(HeadlessGraphicsEnvironment.java:94)
	at FontTest.main(FontTest.java:7)
```

Just installing the DejaVu fonts (because that's what specified in the files shipped with Zulu) does not help. You have to install fontconfig, too (see https://support.azul.com/hc/en-us/articles/360034030692-Using-Fonts-with-OpenJDK-Zulu). These are the packages that are pulled in when installing fontconfig on Ubuntu:

```
fontconfig-config libexpat1 libfontconfig1 libfreetype6 libpng16-16 ucf
```

(Everything tested on Ubuntu 20.04)

**Comment by aahlenst** [comment]
Perfect is the enemy of good, so let's do the following.

Recommend people that use the binary to install fontconfig + DejaVu fonts:

* Debian/Ubuntu/Mint: `sudo apt install fonts-dejavu fontconfig`
* RHEL/CentOS/Fedora: `sudo yum install fontconfig dejavu-sans-fonts dejavu-serif-fonts`
* SLES/OpenSUSE: `sudo zypper install dejavu-fonts fontconfig`

I'll merge in the PR for the installers. If anything breaks because of the redundant freetype, we'll sort it out if it happens. I'll also put up a blog post that explains how to deal with the fonts.

Should I open a issue in the build repo about removing freetype from the binary distributions?

The other option would be to bundle the DejaVu fonts. But I don't think this is the right direction.

**Comment by karianna** [comment]
Thanks for digging into this, I think we have a couple of paths forward here for sure.  Let's discuss at the TSC meeting - hopefully, we have some download numbers on installers vs raw binaries as well

**Comment by karianna** [comment]
We see-sawed back and forth and we're back to providing a fontconfig.properties file to remove the NPE and maybe try to add a free font to remove one other class of problems.

**Comment by aahlenst** [comment]
@karianna To save you some time, https://dejavu-fonts.github.io/ is the home of the DejaVu fonts, https://dejavu-fonts.github.io/License.html is their license. The [tarball of Azul Zulu](https://cdn.azul.com/zulu/bin/zulu8.46.0.19-ca-jdk8.0.252-linux_x64.tar.gz) contains some fontconfig.properties templates with appropriate mappings.

**Comment by madhu-broad** [comment]
https://github.com/AdoptOpenJDK/openjdk-support/issues/70#issuecomment-644061750

AdoptOpenJDK/openjdk-support Getting NULL pointer exception at sun.awt.FcFontManager.getDefaultPlatformFont(FcFontManager.java:76)#80

To just to update, We had raised this issue on Solaris platform (both x86 and sparc). Hope the fix will work for Solaris installers and binaries too.

**Comment by karianna** [comment]
Also see https://github.com/AdoptOpenJDK/openjdk-build/issues/693

**Comment by madhu-broad** [comment]
Hi, 

Any update on this, Can we expect something by this month end or early October?

**Comment by aahlenst** [comment]
@madhu-broad Self-service instructions: https://github.com/AdoptOpenJDK/openjdk-support/issues/70#issuecomment-652482762

**Comment by aahlenst** [comment]
I did some further research:

* Genuine OpenJDK builds (jdk.java.net as well as the builds provided by Red Hat) don't come with Freetype **on Linux**.
* https://docs.oracle.com/javase/9/intl/font-configuration-files.htm states that Windows is the last platform that OpenJDK provides font configuration for (macOS does not support font configuration anyway, support for Linux and Solaris was discontinued). The reason: "(...) Oracle JDK is moving away from providing custom font configuration files on Linux platforms, as they are difficult to keep up to date across distributions and versions."

Without Freetype, you get an error like:

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: /root/jdk-11/lib/libfontmanager.so: libfreetype.so.6: cannot open shared object file: No such file or directory
        at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
        at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2430)
        at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2487)
        at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2684)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2638)
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:829)
        at java.base/java.lang.System.loadLibrary(System.java:1867)
        at java.desktop/sun.font.FontManagerNativeLibrary$1.run(FontManagerNativeLibrary.java:57)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.desktop/sun.font.FontManagerNativeLibrary.<clinit>(FontManagerNativeLibrary.java:32)
        at java.desktop/sun.font.SunFontManager$1.run(SunFontManager.java:270)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.desktop/sun.font.SunFontManager.<clinit>(SunFontManager.java:266)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:398)
        at java.desktop/sun.font.FontManagerFactory$1.run(FontManagerFactory.java:82)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.desktop/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:74)
        at java.desktop/sun.java2d.SunGraphicsEnvironment.getFontManagerForSGE(SunGraphicsEnvironment.java:189)
        at java.desktop/sun.java2d.SunGraphicsEnvironment.getAvailableFontFamilyNames(SunGraphicsEnvironment.java:223)
        at java.desktop/sun.java2d.SunGraphicsEnvironment.getAvailableFontFamilyNames(SunGraphicsEnvironment.java:251)
        at java.desktop/sun.java2d.HeadlessGraphicsEnvironment.getAvailableFontFamilyNames(HeadlessGraphicsEnvironment.java:75)
        at FontTest.main(FontTest.java:7)
```

Install the packages as recommended in https://github.com/AdoptOpenJDK/openjdk-support/issues/70#issuecomment-652482762 and the problem is gone and you get proper integration with the operating system.

To sump up: I think we would get ourselves into trouble by providing font configuration or even fonts. Providing fonts and font configuration is not a problem to be tackeld on the level of the JDK but on the level of the operating system.

**Comment by aahlenst** [comment]
TSC decided not to bundle any fonts and to relegate the font handling to the operating system for the reasons documented in https://github.com/AdoptOpenJDK/openjdk-support/issues/70#issuecomment-703103511.

We're going to publish a blog post that explains how to install the fonts (https://github.com/AdoptOpenJDK/blog/issues/498). We have instructions for Linux in https://github.com/AdoptOpenJDK/openjdk-support/issues/70#issuecomment-652482762. We need instructions for Solaris/AIX. We're also going to update the migration guide (https://github.com/AdoptOpenJDK/openjdk-website/issues/799). 

Later on, we can decide whether we remove Freetype from the binary builds (https://github.com/AdoptOpenJDK/openjdk-build/issues/2133). Probably best to do this together with the move to Eclipse.
